modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/granulecollector/ReprojectingSubmosaicProducerFactory.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * GeoTools - The Open Source Java GIS Toolkit
+ * http://geotools.org
+ *
+ * (C) 2016, Open Source Geospatial Foundation (OSGeo)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation;
+ * version 2.1 of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ */
+
+package org.geotools.gce.imagemosaic.granulecollector;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.geotools.gce.imagemosaic.RasterLayerRequest;
+import org.geotools.gce.imagemosaic.RasterLayerResponse;
+import org.geotools.gce.imagemosaic.RasterManager;
+
+/**
+ * Creates a SubmosaicProducer that can handle reprojecting granules into the mosaic's target CRS
+ */
+public class ReprojectingSubmosaicProducerFactory implements SubmosaicProducerFactory {
+    @Override
+    public List<SubmosaicProducer> createCollectors(RasterLayerRequest request,
```

Comment:
Makes it look like heterogeneous crs mosaic are not going to support stacking merge behavior on ND mosaics correctly (since the default  submosaic producer has special handling for it and will return multiple submosaic producers for it, but not this one)